st.from_types with Union and Protocol weirdly requires int not to throw

I've managed to extract this MWE from my buggy tests

```python
from typing import ClassVar, Dict, ForwardRef, Protocol, Sequence, TypeAlias, Union

from hypothesis import given, settings
from hypothesis import strategies as st


class IsDataclass(Protocol):
    __dataclass_fields__: ClassVar[Dict]


Extended: TypeAlias = Union[Sequence["Extended"], int, bool, bytes, IsDataclass]

st.register_type_strategy(IsDataclass, st.just(b""))
st.register_type_strategy(
    ForwardRef("Extended"),
    st.recursive(st.one_of(st.integers(), st.booleans(), st.binary()), st.tuples),
)


@given(a=...)
@settings(max_examples=1000)
def test_extended(a: Extended):
    print(a)
```

Defining `Extended` without `int`, ie just like

```
Extended: TypeAlias = Union[Sequence["Extended"], bool, bytes, IsDataclass]
```

will crash with

```
E           TypeError: Instance and class checks can only be used with @runtime_checkable protocols
```

I'm using python3.10 and from my `uv.lock` I see

```
[[package]]
name = "hypothesis"
version = "6.112.1"
source = { registry = "https://pypi.org/simple" }
dependencies = [
    { name = "attrs" },
    { name = "exceptiongroup", marker = "python_full_version < '3.11'" },
    { name = "sortedcontainers" },
]
```

--- a/hypothesis-python/RELEASE.rst
+++ b/hypothesis-python/RELEASE.rst
@@ -1,0 +1,3 @@
+RELEASE_TYPE: patch
+
+This patch fixes a bug where :func:`~hypothesis.strategies.from_type` would error on certain types involving :class:`~python:typing.Protocol` (:issue:`4194`).

--- a/hypothesis-python/src/hypothesis/strategies/_internal/types.py
+++ b/hypothesis-python/src/hypothesis/strategies/_internal/types.py
@@ -541,7 +541,9 @@ def from_typing_type(thing):
         else:
             union_elems = ()
         if not any(
-            isinstance(T, type) and issubclass(int, get_origin(T) or T)
+            # see https://github.com/HypothesisWorks/hypothesis/issues/4194 for
+            # try_issubclass.
+            isinstance(T, type) and try_issubclass(int, get_origin(T) or T)
             for T in [*union_elems, elem_type]
         ):
             mapping.pop(bytes, None)

--- a/hypothesis-python/tests/cover/test_lookup.py
+++ b/hypothesis-python/tests/cover/test_lookup.py
@@ -1237,3 +1237,21 @@ def test_from_type_resolves_required_posonly_args(n: CustomInteger):
     # st.builds() does not infer for positional arguments, but st.from_type()
     # does.  See e.g. https://stackoverflow.com/q/79199376/ for motivation.
     assert isinstance(n, CustomInteger)
+
+
+class MyProtocol(typing.Protocol):
+    pass
+
+
+def test_issue_4194_regression():
+    # this was an edge case where we were calling issubclass on something
+    # that was not a type, which errored. I don't have a more principled test
+    # case or name for this.
+    inner = typing.Union[typing.Sequence["A"], MyProtocol]
+    A = typing.Union[typing.Sequence[inner], MyProtocol]
+
+    with (
+        temp_registered(MyProtocol, st.just(b"")),
+        temp_registered(typing.ForwardRef("A"), st.integers()),
+    ):
+        find_any(st.from_type(A))

